File: LV1/문자열 나누기.js
function solution(s) {
  let count = 0;
  let index = 0;

  while (index < s.length) {
    let firstChar = s[index];
    let firstCharCount = 1;
    let otherCharCount = 0;
    index++;

    while (index < s.length) {
      if (s[index] === firstChar) {
        firstCharCount++;
      } else {
        otherCharCount++;
      }

      if (firstCharCount === otherCharCount) {
        count++;
        index++;
        break;
      }

      index++;
    }
    if (firstCharCount !== otherCharCount) {
      count++;
    }
  }

  return count;
}

File: LV1/문자열 나누기.java
class Solution {

  public int solution(String s) {
    int answer = 0;
    int count = 0;
    int index = 0;

    while (index < s.length()) {
      char firstChar = s.charAt(index);
      int firstCharCount = 1;
      int otherCharCount = 0;
      index++;

      while (index < s.length()) {
        if (s.charAt(index) == firstChar) {
          firstCharCount++;
        } else {
          otherCharCount++;
        }

        if (firstCharCount == otherCharCount) {
          count++;
          index++;
          break;
        }

        index++;
      }
      if (firstCharCount != otherCharCount) {
        count++;
      }
    }

    return count;
  }
}
